components/ws-daemon/pkg/content/initializer.go

```diff
@@ -390,13 +389,28 @@ func (rs *remoteContentStorage) Download(ctx context.Context, destination string
 	}
 
 	span.SetTag("URL", info.URL)
-	resp, err := http.Get(info.URL)
+
+	tempFile, err := os.CreateTemp("", "remote-content-")
+	if err != nil {
+		return true, xerrors.Errorf("cannot create temporal file: %w", err)
+	}
+	defer tempFile.Close()
+	defer os.Remove(tempFile.Name())
+
+	args := []string{
+		"-x16", "-j12",
```

Comment:
Do we need to consider a self-hosted env?

Reply:
What do you mean?

Reply:
Sorry for the lack of information :bow:  I think self-hosted environments may require less CPU cores than SaaS. I found it difficult to judge whether this degree of parallelism is appropriate at that time.

Reply:
We have the same sizes now as the default. We could choose values using the CPU core count, but I would prefer to do that in a different PR after we use it in SaaS and gather metrics.

Reply:
:+1: